### ISSUE TITLE
DragonFruit fails with assembly names that contain whitespace

### ISSUE THREAD

**Issue by danielchalmers** [dotnet/command-line-api#684]
If you have an assembly name containing whitespace, DragonFruit fails with `System.ArgumentException: RootCommand alias cannot contain whitespace`.

**Repro:**

1. Create a C# .NET Core 3.0 Console App named `Hello DragonFruit`
2. Add to `Hello DragonFruit.csproj`: `<PackageReference Include="System.CommandLine.DragonFruit" Version="0.3.0-alpha.19405.1" />`
3. Add to `Program.cs`: `static void Main(string dragonFruit) => Console.WriteLine($"Hello {dragonFruit}!");`
4. Launch it: `"Hello DragonFruit.exe" --dragon-fruit "DragonFruit"`

**Error:**

```
Unhandled exception. System.ArgumentException: RootCommand alias cannot contain whitespace: "Hello DragonFruit"
   at System.CommandLine.Symbol.AddAliasInner(String alias)
   at System.CommandLine.Symbol..ctor(IReadOnlyCollection`1 aliases, String description)
   at System.CommandLine.Command..ctor(String name, String description)
   at System.CommandLine.RootCommand..ctor(String description)
   at System.CommandLine.Builder.CommandLineBuilder..ctor(Command rootCommand)
   at System.CommandLine.DragonFruit.CommandLine.BuildParser(MethodInfo method, String xmlDocsFilePath, Object target)
   at System.CommandLine.DragonFruit.CommandLine.InvokeMethodAsync(String[] args, MethodInfo method, String xmlDocsFilePath, Object target, IConsole console)
   at System.CommandLine.DragonFruit.CommandLine.ExecuteAssemblyAsync(Assembly entryAssembly, String[] args, String entryPointFullTypeName, String xmlDocsFilePath, IConsole console)
   at AutoGeneratedProgram.Main(String[] args) in C:\Users\example\source\repos\Hello DragonFruit\Hello DragonFruit\obj\Debug\netcoreapp3.0\Hello DragonFruit.Program.g.cs:line 10
   at AutoGeneratedProgram.<Main>(String[] args)
```

**Comment by KathleenDollard** [comment]
Interesting. 

Can you explain a bit more why you want to do this? Where would a space in an exe name be desirable?

**Comment by danielchalmers** [comment]
@KathleenDollard I mean, why not right? I don't think it's that uncommon to have spaces in assembly names, and every other part of the toolchain supports it.

A new developer with a "Hello World" project on a default template wouldn't be able to use the library without renaming their project, adding a bit to the learning curve.

**Comment by jonsequitur** [comment]
What would be an expected behavior?

A space in an assembly name isn't uncommon, but a space in an executable name is a bit of a usability barrier, and a space in a subcommand or option is a larger one. I'm not aware of a command line parser that supports the latter, which is why we don't support aliases with spaces.

**Comment by danielchalmers** [comment]
@jonsequitur 

>A space in an assembly name isn't uncommon, but a space in an executable name is a bit of a usability barrier

Agreed, but It's the assembly name that does it. Even if the exe contains no whitespace, it'll still fail.

Whitespace in options is one thing, but I think in the `RootCommand` is fair. If you parse commands through `string[] args` there's no problem with having whitespace in the assembly name, but DragonFruit refuses it. It's one barrier to a drop-in replacement.

**Comment by jonsequitur** [comment]
I have a simple fix in #688. It unblocks this issue somewhat tactically by removing whitespace from root command names that are inferred from the name of the executable.
